pymc_marketing/clv/utils.py

```diff
@@ -244,13 +248,13 @@ def clv_summary(
         Column in the transactions DataFrame that denotes the customer_id.
     datetime_col:  string
         Column in the transactions DataFrame that denotes the datetime the purchase was made.
-    monetary_value_col: string, optional
+    monetary_value_col: Optional[str], by default None
         Column in the transactions DataFrame that denotes the monetary value of the transaction.
         Optional; only needed for spend estimation models like the Gamma-Gamma model.
-    observation_period_end: datetime, optional
-         A string or datetime to denote the final date of the study.
-         Events after this date are truncated. If not given, defaults to the max 'datetime_col'.
-    datetime_format: string, optional
+    observation_period_end: Optional[Union[str, pd.Period, datetime]], by default None
```

Comment:
In PyMC we don't usually bother to do this, since it's duplicated work and docstrings can become much more verbose. We usually have type-hints on the function call, and readable type descriptions in the docstrings.

```suggestion
    observation_period_end: Optional[Union[str, pd.Period, datetime]], by default None
```

No strong preference, but I like less work rather than more :)

Reply:
I think as long we have a convention we can be flexible, shall I undo the doctring changes?

Reply:
Yeah

Reply:
reverted them inhttps://github.com/pymc-labs/pymc-marketing/pull/248/commits/07d923ea717534f3657d49a31bf82e6b9346f33a